baseplate/lib/edge_context.py

```diff
@@ -81,9 +81,21 @@ def oauth_client_id(self) -> Optional[str]:
     def oauth_client_type(self) -> Optional[str]:
         raise NotImplementedError
 
+    @property
+    def scopes(self) -> Set[str]:
+        raise NotImplementedError
+
+    @property
+    def loid(self) -> Optional[str]:
+        raise NotImplementedError
+
+    @property
+    def loid_created_ms(self) -> Optional[int]:
+        raise NotImplementedError
+
 
 class ValidatedAuthenticationToken(AuthenticationToken):
-    def __init__(self, payload: Dict[str, str]):
+    def __init__(self, payload: Dict[str, Any]):
```

Comment:
was this wrong before? it looks like `payload[roles]` is a `List[str]`, not `str`

Reply:
yea i dont know why before set as str